### ISSUE TITLE
Remove warnings output from tests

### ISSUE THREAD

**Issue by xsawyerx** [PerlDancer/Dancer2#1265]
The tests output various things to STDERR. I want to clean up that. It's annoying.

**Comment by racke** [comment]
:+1: 

Also I started to use Test::Warnings in my test scripts, so unwanted warning are catched as well.

**Comment by xsawyerx** [comment]
We can use that in all the scripts. We can also use Capture::Tiny. I remember Test::Warnings had some issues, but I guess those are resolved since.

**Comment by veryrusty** [comment]
Hmmm `prove -lvr t/ 2>stderr.txt` results in four lines of text (damn case insensitive filesystem) from `t/dsl/send_as.t`:
```
Subroutine from_json redefined at lib/Dancer2/Serializer/jSoN.pm line 13.
Subroutine to_json redefined at lib/Dancer2/Serializer/jSoN.pm line 15.
Subroutine serialize redefined at lib/Dancer2/Serializer/jSoN.pm line 18.
Subroutine deserialize redefined at lib/Dancer2/Serializer/jSoN.pm line 31
```

Are you sure the annoying output is to STDERR?

**Comment by xsawyerx** [comment]
@veryrusty Good point. Let me rephrase: Remove all output not part of TAP (unless "note" or "diag").

**Comment by xsawyerx** [comment]
I'm not seeing any warnings or bad output right now, so I'm closing this ticket until I have something concrete/substantial.
